les/server_handler.go

```diff
@@ -273,7 +276,9 @@ func (h *serverHandler) handleMsg(p *peer) error {
 		}
 		query := req.Query
 		if accept(req.ReqID, query.Amount, MaxHeaderFetch) {
+			p.wg.Add(1)
```

Comment:
This could be a part of the `accept` function. You can leave it like this though if you think it improves readability.